lib/pure/uri.nim

```diff
@@ -165,18 +165,21 @@ iterator decodeQuery*(data: string): tuple[key, value: TaintedString] =
   ## data consists of.
   runnableExamples:
     import std/sugar
-    let s = collect(newSeq):
-      for k, v in decodeQuery("foo=1&bar=2"): (k, v)
-    doAssert s == @[("foo", "1"), ("bar", "2")]
+    let s = collect:
+      for k, v in decodeQuery("foo=1&bar=2=3"): (k, v)
+    doAssert s == @[("foo", "1"), ("bar", "2=3")]
 
-  proc parseData(data: string, i: int, field: var string): int =
+  proc parseData(data: string, i: int, field: var string, sep: char): int =
     result = i
     while result < data.len:
-      case data[result]
+      let c = data[result]
+      case c
       of '%': add(field, decodePercent(data, result))
       of '+': add(field, ' ')
-      of '=', '&': break
-      else: add(field, data[result])
+      of '&': break
+      else:
+        if c == sep: break
+        else: add(field, data[result])
```

Comment:
```suggestion
      elif c == sep: break
      else: add(field, data[result])
```

Reply:
This is the else of the case statement, I'm not convinced that elif is a valid option for it.

Reply:
it works, i verified